src/NUnitFramework/framework/Interfaces/ITest.cs

```diff
@@ -94,6 +94,12 @@ public interface ITest : IXmlNodeBuilder
         ITest Parent { get; }
 
         /// <summary>
+        /// Gets the name of the TestWorker that
+        /// is executing this test.
+        /// </summary>
+        string WorkerId {get;}
+
+        /// <summary>
```

Comment:
Starting at the top, this is the first thing I notice. I don't like adding info to a test about how it is being executed. That breaks the separation we tried to create so carefully by having a WorkItem, which knows about tests. Let's see where you go with this as I read further.